src/components/MapView/Legends/index.tsx

```diff
@@ -27,14 +28,26 @@ function Legends({ classes, layers }: LegendsProps) {
   const isAnalysisLayerActive = useSelector(isAnalysisLayerActiveSelector);
 
   const legendItems = [
-    ...layers.map(({ title, legend, legendText }) => {
-      if (!legend || !legendText) {
+    ...layers.map(layer => {
+      if (!layer.legend || !layer.legendText) {
         // this layer doesn't have a legend (likely boundary), so lets ignore.
         return null;
       }
+
+      const wmsParams =
```

Comment:
Add a comment to explain what we are doing here.

Reply:
included in c3febf90894666e7c7503d4e009a498905b5cdc0